src/dataframerow/dataframerow.jl

```diff
@@ -62,7 +62,11 @@ Base.@propagate_inbounds function DataFrameRow(sdf::SubDataFrame, row::Integer,
         throw(BoundsError("attempt to access a data frame with $(nrow(sdf)) " *
                           "rows at index $row"))
     end
-    colindex = SubIndex(index(parent(sdf)), parentcols(index(sdf), cols))
+    colindex = if index(sdf) isa Index
```

Comment:
I don't quite understand the change here.

Reply:
The consequence of the change: if you do:
```
sdf = @view df[rows, :]
dfr = sdf[row, :]
```
then this has the same result as:
```
dfr = df[row, :]
```
In other words if you keep using `:` in `view` creation then the `:`-related property propagates.

The major consequence is that we have an improved performance of `DataFrameRow` creation from `SubDataFrame` (as we avoid creation of `SubIndex`). Additionally this makes the behavior consistent with the documentation (where we say that if you use `:` for column selection then view` follows changes of columns in the parent).

The practical way to think about it is the following: if you use `:` as column selector then view inherits `Index` from parent; if you use something else then `:` then in view we create a `SubIndex`. The original reason for this distinction was ensuring performance of operations like `eachrow` (where you know you select all columns always).

Reply:
Actually your comment made me realize that we should have the same optimization in nested `SubDataFrame` (which I now added). Thank you.